src/lsm/schema.zig

```diff
@@ -499,13 +499,12 @@ pub const TableData = struct {
         data_block: BlockPtrConst,
     ) []align(16) const u8 {
         const header = header_from_block(data_block);
-        // TODO we should be able to cross-check this with the header size
-        // for more safety.
         const used_values: u32 = header.request;
         assert(used_values > 0);
         assert(used_values <= schema.value_count_max);
 
         const used_bytes = used_values * schema.value_size;
+        assert(@sizeOf(vsr.Header) + used_bytes == header.size);
```

Comment:
Added. It's not as useful as I thought though because if we calculate the real padding it would end up being a (syntactic) tautology. But it's still good to check.